src/frequenz/sdk/actor/power_distributing/_battery_status.py

```diff
@@ -0,0 +1,364 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Tool to track information whether battery is working or not."""
+
+import asyncio
+import datetime
+import logging
+from typing import Dict, Optional, Set, Union
+
+import pytz
+from frequenz.channels import Receiver
+
+from ...microgrid import ComponentGraph
+from ...microgrid import get as get_microgrid
+from ...microgrid.component import (
+    BatteryData,
+    Component,
+    ComponentCategory,
+    InverterData,
+)
+from ...microgrid.component._component_errors import ErrorLevel
+from ...microgrid.component._component_states import (
+    BatteryRelayState,
+    BatteryState,
+    InverterState,
+)
+from ._battery_status_utils import BatInvReceivers, BlockedComponents
+from .result import PartialFailure, Result, Success
+
+_logger = logging.getLogger(__name__)
+
+
+class BatteryStatus:
```

Comment:
To me this looks like a collection of methods belonging to two different categories: ones that work on a single component, and ones that work on a collection of components.

So, I'd suggest splitting this into two classes for cleaner layers:
`BatteryPool`: has methods that work on a collection of battery IDs
`BatteryUtils`: has functions for  a single battery ID, like whether it's online or not, what is it's connected inverter, etc.

Reply:
done